src/lib/utils/meta_table_manager.cpp

```diff
@@ -2,18 +2,83 @@
 
 #include "constant_mappings.hpp"
 #include "hyrise.hpp"
+#include "resolve_type.hpp"
 #include "statistics/table_statistics.hpp"
 #include "storage/base_encoded_segment.hpp"
+#include "storage/segment_iterables/any_segment_iterable.hpp"
 #include "storage/table.hpp"
 #include "storage/table_column_definition.hpp"
 
+namespace {
+
+using namespace opossum;  // NOLINT
+
+int32_t get_distinct_value_count(const std::shared_ptr<BaseSegment> segment) {
```

Comment:
Why does this return a signed integer, can it ever become positive?

Reply:
Nope, but since we only use this method within the meta table manager to fill an integer columns (and we don't have unsigned column types), I don't think the additional cast would benefit anyone.

Reply:
Hm, I don't know. I could easily imagine this method to be used elsewhere in the future and I had the feeling that we are quite strict with our return types so far.

Reply:
The dictionary handling required another cast, so now returning size_t has the overall less casts. Changed. :)